src/main/java/updater/UpdateManager.java

```diff
@@ -123,6 +157,57 @@ private boolean extractJarUpdater() {
     }
 
     /**
+     * Keeps the number of HT Jar in the folder used as backup to specified amount.
+     */
+    private void cleanupHTBackupJar() {
+        logger.info("Cleaning up backup JAR");
+
+        File currDirectory = new File(".");
+
+        File[] filesInCurrentDirectory = currDirectory.listFiles();
+
+        if (filesInCurrentDirectory == null) {
+            // current directory always exists
+            assert false;
+            return;
+        }
+        assert filesInCurrentDirectory != null;
+
+        List<File> listOfFilesInCurrDirectory = Arrays.asList(filesInCurrentDirectory);
+
+        List<File> allHtBackupFiles = listOfFilesInCurrDirectory.stream()
+                .filter(f -> !f.getName().equals(String.format("HubTurbo_%s.jar", Version.getCurrentVersion())) &&
+                             f.getName().matches(HT_BACKUP_FILENAME_PATTERN_STRING))
+                .sorted(getHtBackupFileComparatorByVersion())
+                .collect(Collectors.toList());
+
+        if (allHtBackupFiles.isEmpty()) {
+            return;
+        }
+
+        for (int i = 0; i < (allHtBackupFiles.size() - MAX_HT_BACKUP_JAR_KEPT); i++) {
+            logger.info("Deleting " + allHtBackupFiles.get(i).getName());
+            if (!allHtBackupFiles.get(i).delete()) {
+                logger.warn("Failed to delete old HT backup file " + allHtBackupFiles.get(i).getName());
+            }
+        }
+    }
+
+    private Comparator<File> getHtBackupFileComparatorByVersion() {
+        return (a, b) -> {
+            Pattern htJarFileBackupPattern = Pattern.compile(HT_BACKUP_FILENAME_PATTERN_STRING);
```

Comment:
Yes, any regex matching should be extracted out and tested separately, if possible.